src/Transit.php

```diff
@@ -68,6 +68,24 @@ public function select(string $domainClass, array $whereEquals = []) : TransitSe
         );
     }
 
+    /**
+     * Attach a domain object to plan.
+     *
+     * @param \Atlas\Transit\object $domain
+     *
+     * @return \Atlas\Transit\Transit
+     * @throws \Atlas\Transit\Exception
+     */
+    public function attach(object $domain): Transit
```

Comment:
The problem with this method: which method should the end user use? `store` or `attach`? This results in an unclear API.

Reply:
Like Doctrine; it's needed to "attach" ("merge" with Doctrine) the entity to Transit before modification.
If the entity is modified and stored, she is already present and an update is done.

Reply:
If you have an another suggestion of method name, i'm ready :)

Reply:
Doctrine no longer works as an example, because `merge` and `detach` are deprecated and will be removed in version 3. 😉 

https://github.com/doctrine/orm/blob/master/UPGRADE.md#bc-break-removed-entitymanagermerge-and-entitymanagerdetach-methods
https://github.com/doctrine/orm/pull/1577#issue-52511497

Reply:
Yes i see that 👍 .
But how do for the current problematic?
It's the job of a framework? or need to integrate a concept into Transit?

Reply:
See also at Doctrine:

> It is a good idea to avoid storing entities in serialized formats such as `$_SESSION`: instead, store the entity identifiers or raw data.

https://github.com/doctrine/orm/blob/master/docs/en/cookbook/entities-in-session.rst#entities-in-the-session

…and this is the path you should go.

Reply:
It's not too similar to `Transit::select(string $domainClass, array $whereEquals = [])` method; with second parameter who accept a scalar or array argument?

Reply:
> It's not too similar to…

You are on the right track. 😉
Don't serialize your domain objects, instead:

* store the entity identifiers or raw data (in session, etc.)
* use the existing `select` method to get your domain object

…and the entire pull request is not needed.